### DISCUSSION TITLE
How to test customised module on SKR?

### DISCUSSION THREAD

**Discussion by ukff** [kyma-project/lifecycle-manager#1161] [Q&A] [answered]
### Question

In BTP Manager i have changes which i want to test before publishing them to public channels/landscapes. Basically i only want install my custom version of BTP Manager on only my one development Kyma cluster.

PR can be found here: https://github.com/kyma-project/btp-manager/pull/517
When you inspect it, it contains a rendered ModuleTemplate with newest image itp. (job is called pull-btp-manager-module-build) 

Could you please explain how to apply module with my still development and not stable changes to only selected SKR since i dont want to publish if anywhere beside my cluster.

I am aware of `remoteModuleTemplateRef`, but more detailed real usage example would be more usefull, than only mention one field in docs

### Should this question be covered by our FAQ?

None

**Comment by ruanxin** [comment]
Here will use [template-operator](https://github.com/kyma-project/template-operator) as an example.

## Steps:

1. Upload customized module template on SKR cluster.
2. Manually enable the module in Kyma CR `spec.modules` field, with `remoteModuleTemplateRef` as an additional field, the value could be one of these:
 - the name of the moduletemplate
 - the short name of the module, in this case: template-operator,
 - or the [fqdn](https://github.com/kyma-project/template-operator/blob/main/module-config.yaml#L1C7-L1C47) of the module.  

Please only use module name if your module has not been submitted to KCP yet, because it will bring conflict due to the same module name.

```
spec:
  modules:
    - name: template-operator
      remoteModuleTemplateRef: kyma-project.io/module/template-operator
```
4. Verify if this module gets enabled successfully.

**Reply by ukff** [reply]
@ruanxin could you provide full template-operator ModuleTemplate YAML to which you refer? I suspect they may be some difference, in our build pipelines which make things not work. I wanted compare it.

**Reply by ruanxin** [reply]
you can create the ModuleTemplate by CLI, we don't release this yaml yet. Also, you can reference ModuleTemplate from other modules, like [keda-manager](https://github.com/kyma-project/keda-manager/releases/tag/1.0.2).

**Comment by janmedrek** [comment, marked as answer]
The functionality for handling Remote Module Templates was turned off, the agreement is that Module Operators should be tested independently from KLM's logic.
